sound/soc/sof/intel/hda-loader.c

```diff
@@ -77,7 +77,7 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 	return hstream->stream_tag;
 
 error:
-	hda_dsp_stream_put_pstream(sdev, hstream->stream_tag);
+	hda_dsp_stream_put(sdev, direction, hstream->stream_tag);
```

Comment:
same here, the second argument is a direction and elsewhere it's substream->stream.

Reply:
for substream->stream,  the definition is: 
int stream;			/* stream (direction) */
so it is direction

Reply:
yes, it is a good idea. I will refine it.